src/init.cpp

```diff
@@ -187,11 +187,6 @@ void Shutdown()
     StopREST();
     StopRPC();
     StopHTTPServer();
-#ifdef ENABLE_WALLET
```

Comment:
Correct - I don't believe the early wallet flush is necessary (although I may be wrong). See PR description:

> The most important one is that Flush() no longer gets called twice on shutdown. I think this is fine, and all the tests pass, but this probably requires some careful consideration from people who are more familiar with the wallet code.